### PR TITLE
Support set/unset property statements via the rest api. 

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

Support set/unset property statements via the rest api.  However, scope the to the current request.  This allows a single request to contain multiple statements interspersed with set/unset statements.

e.g.

```sql
SET 'some.prop'='some.value';
CREATE TABLE Foo ...
SET 'some.prop'='some.other.value';
CREATE TABLE Bar ...
UNSET 'some.prop';
CREATE TABLE Zar ...
```

### Testing done 
Usual.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java [hunk 1]
```diff
@@ -760,27 +765,98 @@ public void shouldReturn5xxOnStatementSystemError() {
...
+    // Then:
+    assertThat(response, instanceOf(KsqlStatementErrorMessage.class));
+    assertThat(response.getErrorCode(), is(Errors.ERROR_CODE_BAD_STATEMENT));
+    assertThat(response.getMessage(),
+        containsString("Invalid value invalid value for configuration ksql.sink.replicas: "
+            + "Not a number of type SHORT"));
+  }
+
+  @Test
+  public void shouldUnsetProperty() {
```
Comment (vcrfxia) [inline ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java]: What's the expected behavior for unsetting a property that wasn't provided/set? From the current implementation it looks like it's a no-op. Perhaps add a test to confirm no error is thrown?

**Comment by miguno** [comment]
FWIW, my personal intuition as a user would be:

- `SET` would apply from the moment it was issued and onwards.
    - This could be confusing for settings that would really query-specific by nature, e.g. if we allowed the user to customize the KStreams `application.id` of a KSQL query.
- A `SET` scoped to one specific query, but not to a subsequent query, would need to be really called out as such.  I could even see that the `WITH` clause would be a good place for query-specific properties, like `CREATE ... WITH ('auto.offset.reset' = 'earliest')`.

And I agree with @rodesai's last comment above (GH doesn't show me a direct link so I can't add it here).

**Comment by big-andy-coates** [comment]
Hey @rodesai 

The CLI side of the equation is already there, right? `Cli.setProperty`...

I'll add a test case to prove this works.

**Comment by big-andy-coates** [comment]
Test case added.

**Comment by rodesai** [comment]
Thanks @big-andy-coates, I forgot the CLI also runs through the command list.

**Pull request merged**